src/contract.rs

```diff
@@ -527,30 +612,54 @@ fn get_bank_transfer_to_msg(recipient: &Addr, denom: &str, native_amount: Uint12
     transfer_bank_cosmos_msg
 }
 
+fn fee_decimal_to_uint128(decimal: Decimal) -> StdResult<Uint128> {
+    let result: Uint128 = decimal
+        .atomics()
+        .checked_mul(FEE_SCALE_FACTOR)
+        .map_err(StdError::overflow)?;
+
+    Ok(result / FEE_DECIMAL_PRECISION)
```

Comment:
Currently this whole operation seems to have the effect of multiplying the raw decimal value by 100. Could we just combine these? Do we need that?

Under the hood `Decimal` has 18 decimal places: https://docs.rs/cosmwasm-std/latest/src/cosmwasm_std/math/decimal.rs.html#31

Reply:
Edit: I misread this. Seems like this is being done so that the user can input a percent between zero and 100. This is slick. :)

Might be nice to add a comment here about how this scaling comes together. Can get a little lost because we divide by 10,000 later.

Reply:
What do you mean by combine? What would the function look like if you were to combine?

Reply:
Ah. Sorry. GitHub did something weird there. Didn't mean to submit the old version of this comment.